source/common/router/config_impl.cc

```diff
@@ -65,6 +67,82 @@ Optional<uint64_t> HashPolicyImpl::generateHash(const Http::HeaderMap& headers)
   return hash;
 }
 
+HeaderFormatterSharedPtr RequestHeaderParser::parseInternal(const std::string& format) {
+	std::string variable_name;
+	if (format.find("%") == 0) {
```

Comment:
Why not just have the general AccessLogFormatParser::parse and then make this a special case? In general, I'm a fan of less code and less boiler plate. If we find a bug in parsing, I rather fix it in one place than two.